Modules/TRD/src/PulseHeightTrackMatch.cxx

```diff
@@ -0,0 +1,308 @@
+    auto tracks = ctx.inputs().get<gsl::span<o2::trd::TrackTRD>>("tracks");
+    auto triggerrecordstracks = ctx.inputs().get<gsl::span<o2::trd::TrackTriggerRecord>>("trigrectrk");
+
+    std::vector<o2::trd::Digit> digitv(digits.begin(), digits.end());
+
+    if (digitv.size() == 0)
+      continue;
+
+    std::vector<unsigned int> digitsIndex(digitv.size());
+    std::iota(digitsIndex.begin(), digitsIndex.end(), 0);
+
+    int nTrgWDigits = 0;
+    int nTotalTrigger = 0;
+    auto crustart = std::chrono::high_resolution_clock::now();
+
+    for (const auto& trigger : triggerrecords) {
+      // printf("-------------Trigger----------------------BCData for Trigger %i, %i \n ",trigger.getBCData().orbit,trigger.getBCData().bc);
+      nTotalTrigger++;
+
+      if (trigger.getNumberOfDigits() == 0) {
+        continue;
+      }
+
+      else {
+        nTrgWDigits++;
+      }
+
+      if (trigger.getNumberOfDigits() > 10000) {
+        mDigitsPerEvent->Fill(9999);
+      } else {
+        mDigitsPerEvent->Fill(trigger.getNumberOfDigits());
+      }
+
+      // now sort digits to det,row,pad
+      std::sort(std::begin(digitsIndex) + trigger.getFirstDigit(), std::begin(digitsIndex) + trigger.getFirstDigit() + trigger.getNumberOfDigits(),
+                [&digitv](unsigned int i, unsigned int j) { return digitIndexCompare_phtm(i, j, digitv); });
+
+      for (const auto& triggerTracks : triggerrecordstracks) {
+
+        if (trigger.getBCData() != triggerTracks.getBCData()) {
+          continue;
+        }
+
+        int nTracks = 0;
+        for (int iTrack = triggerTracks.getFirstTrack(); iTrack < triggerTracks.getFirstTrack() + triggerTracks.getNumberOfTracks(); iTrack++) {
+          nTracks++;
+          const auto& track = tracks[iTrack];
+
+          // printf("Got track with %i tracklets \n", track.getNtracklets());
+          int ntracklets = track.getNtracklets();
+          if (ntracklets > 10)
+            mTrackletsPerMatchedTrack->Fill(9);
+          else
+            mTrackletsPerMatchedTrack->Fill(ntracklets);
+
+          for (int iLayer = 0; iLayer < 6; iLayer++) {
+            int trackletIndex = track.getTrackletIndex(iLayer);
+
+            if (trackletIndex == -1) {
+              continue;
+            }
+
+            const auto& trklt = tracklets[trackletIndex];
+
+            int det = trklt.getDetector();
+            int sec = det / 30;
+            int row = trklt.getPadRow();
+            // int col = trklt.getPadCol();
+
+            // obtain pad number relative to MCM center
+            int padLocal = trklt.getPositionBinSigned() * o2::trd::constants::GRANULARITYTRKLPOS;
+            // MCM number in column direction (0..7)
+            int mcmCol = (trklt.getMCM() % o2::trd::constants::NMCMROBINCOL) + o2::trd::constants::NMCMROBINCOL * (trklt.getROB() % 2);
+            int col = 6.f + mcmCol * ((float)o2::trd::constants::NCOLMCM) + padLocal;
```

Comment:
I remember a discussion with Felix that these calculations should be provided centrally, either in the tracklet class or as a helper function. Can you check with him?